### ISSUE TITLE
The input string '128.000000' was not in a correct format.

### ISSUE THREAD

**Issue by FurStaku** [Wildenhaus/IndexV2#11]
Installed .NET Versions: `6.0, 7.0, 8.0, 9.0`
Version: `v0.4.0.0`
H2A Path: `G:\...\...\...\Halo The Master Chief Collection\halo2`
Steps to Reproduce:
```
Select H2A
Choose a model from the list "Template (Model)".
The error occurs when attempting to open most models in H2A.
```
Error:
```
System.FormatException: The input string '128.000000' was not in a correct format.
   at System.Number.ThrowFormatException[TChar](ReadOnlySpan`1 value)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.ParseValue(StringReader reader, Type type, String value)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.ReadProperty(StringReader reader, T obj)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.OnDeserialize(StringReader reader, T obj)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.Deserialize(StringReader reader)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.LibSaber.Halo2A.Serialization.Scripting.ITextScriptingSerializer.Deserialize(StringReader reader)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.ParseValue(StringReader reader, Type type, String value)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.ReadProperty(StringReader reader, T obj)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.OnDeserialize(StringReader reader, T obj)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.Deserialize(StringReader reader)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.LibSaber.Halo2A.Serialization.Scripting.ITextScriptingSerializer.Deserialize(StringReader reader)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.ParseValue(StringReader reader, Type type, String value)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.ReadProperty(StringReader reader, T obj)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.OnDeserialize(StringReader reader, T obj)
   at LibSaber.Halo2A.Serialization.Scripting.TextScriptingSerializer`1.OnDeserialize(Stream stream, T obj)
   at LibSaber.Halo2A.Serialization.Scripting.ScriptingSerializer`1.Deserialize(Stream stream)
   at Index.Profiles.Halo2A.Jobs.LoadGeometryTexturesJob.GatherAdditionalTextures()
   at Index.Profiles.Halo2A.Jobs.LoadGeometryTexturesJob.OnExecuting()
   at Index.Jobs.JobBase.Execute()
```

**Comment by Wildenhaus** [comment]
Can you list some of the models you're seeing this on? I downloaded a fresh copy of the tool and H2A and was not getting this.

**Comment by FurStaku** [comment]
Thanks for the reply. I’m seeing this error with almost every model I try to open. The only models that worked for me so far are:
- shared/elite_fp_body__h
- shared/fp_arms__h
- 01a_tutorial/sm_geom_00006

Everything else gives me the same error, but the number in the string changes from model to model. I’m not sure why these specific models work, but most others don’t. Any idea what could be causing this?

**Comment by Wildenhaus** [comment]
The only things I can think of is either an old version of Index.Profiles.Halo2A.dll, something odd with your game files, or some breaking change with .NET 9 which is still in pre-release stages. The former would probably cause an error on startup though...

I installed .NET 9 rc1 and still haven't been able to get anything to break.

**Comment by FurStaku** [comment]
OK, I did a clean install of MCC on Steam, uninstalled .NET, and installed only version 8.0.8. I also did a fresh install of the Index tool, but the issue is still persisting. Could this be something else? Should I try compiling the tool myself?

**Comment by Wildenhaus** [comment]
Wow, okay, so I didn't know this, but are you from Europe/Computer is set to a non-english language? Apparently a lot of european countries use a comma as decimal separators rather than a period. I think that's what's happening here.

**Comment by FurStaku** [comment]
Yeah, I’m from the EU, and my PC is set to German. I can switch to English to test if that changes anything and see if it confirms the theory.

**Comment by FurStaku** [comment]
Yeah, I switched my PC to English, and now everything works in H2A ~ so theory confirmed!
It’s strange, though, that this only happens in H2A, since everything worked fine in HCE.

**Comment by Wildenhaus** [comment]
C#'s runtime defaults to whatever language your system is set to. I have a new release with a few other fixes on the way.

**Comment by Wildenhaus** [comment]
[Index-v0.4.0.1.zip](https://github.com/user-attachments/files/17135068/Index-v0.4.0.1.zip)

mind testing this build with your pc set to german?

**Comment by FurStaku** [comment]
Sure! I’ll test this build with my PC set to German and let you know how it goes.

**Comment by FurStaku** [comment]
It works perfectly fine now! Thanks for your help!

**Comment by Wildenhaus** [comment]
Fixed in [v0.4.0.1](https://github.com/Wildenhaus/IndexV2/releases/tag/v0.4.0.1)
